src/components/vertical-menu/vertical-menu-item.component.tsx

```diff
@@ -0,0 +1,140 @@
+import React, { useState, useContext } from "react";
+import { PaddingProps } from "styled-system";
+
+import { filterStyledSystemPaddingProps } from "../../style/utils";
+import { IconType } from "../icon";
+import VerticalMenuFullScreenContext from "./vertical-menu-full-screen.context";
+import {
+  StyledVerticalMenuItem,
+  StyledTitle,
+  StyledAdornment,
+  StyledList,
+  StyledChevronIcon,
+  StyledTitleIcon,
+} from "./vertical-menu.style";
+
+const MenuItemContext = React.createContext({
+  level: 0,
+});
+
+export interface VerticalMenuItemProps<T = React.ElementType>
+  extends PaddingProps {
+  /** Children of the menu item - another level of VerticalMenuItems */
+  children?: React.ReactNode;
+  /** Title of the menu item */
+  title: string;
+  /** Adornment of the menu item meant to be rendered on the right side */
+  adornment?: React.ReactNode | ((isOpen: boolean) => React.ReactNode);
+  /** Icon meant to be rendered on the left side */
+  iconType?: IconType;
+  /** Wether the menu item is active or not */
+  active?: boolean | ((isOpen: boolean) => boolean);
+  /** Height of the menu item */
+  height?: string;
+  /**  Href, when passed the menu item will be rendered as an anchor tag */
+  href?: string;
+  /** Optional component to render instead of the default div, useful for rendering router link components */
+  component?: T;
+}
+
+type InferredComponentProps<T extends React.ElementType> = Omit<
+  React.ComponentProps<T>,
+  keyof VerticalMenuItemProps<T>
+>;
+
+export const VerticalMenuItem = <T,>({
+  title,
+  iconType,
+  adornment,
+  children,
+  component,
+  active,
+  height = "56px",
+  ...rest
+}: T extends React.ElementType
+  ? InferredComponentProps<T> & VerticalMenuItemProps<T>
+  : VerticalMenuItemProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const handleKeyDown = (e: React.KeyboardEvent) => {
```

Comment:
Do we need to support passing `onKeyDown` and `onClick` as props as well like we do in menu-item?

Reply:
According to the requirements VerticalMenu is initially meant to be used only as a navigation. This can be achieved by either passing a `href` prop or by passing custom link component. Considering that there's no need for `onKeyDown` or `onClick` props for now. They might be added in the future if requested.

Reply:
No problem thanks for clarifying

Reply:
FYI: I've changed it to render as `button` when it has children. So now it is possible to trigger the open state by both enter and space keypress (and click obviously)